fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/api/infra/FenixAPIFromExternalServer.java

```diff
@@ -127,9 +126,11 @@ private static JsonObject getInformation(String url, String filename) {
         return infoJson;
     }
 
-    public static String getCanteen(String daySearch) {
-        getInformation();
+    public static String getCanteen(String daySearch, String canteenName) {
```

Comment:
Your shouldn't be changing API since we are not expecting a major version any time soon. Just keep the old method returning the default, and make a new one with the extra parameter.